src/helpers/utils.js

```diff
@@ -0,0 +1,35 @@
+import readline from 'readline';
+
+export async function readLineAsync(query) {
+  return new Promise((resolve, reject) => {
+    if (arguments.length !== 1) {
+      reject(new Error('arguments must be 1'));
+    }
+
+    if (typeof query !== 'string') {
+      reject(new Error('query must be string'));
+    }
+
+    const rl = readline.createInterface({
+      input: process.stdin,
+      output: process.stdout,
+    });
+
+    rl.question(query, input => {
+      rl.close();
+      resolve(input);
+    });
+  });
+}
+
+export async function retryUntilSuccess(callbackFn) {
+  try {
+    return await callbackFn();
+  } catch {
+    return await retryUntilSuccess(callbackFn);
```

Comment:
여기엔 await이 없어도 될 것 같아요!
혹시 왜 callbackFn에는 await이 필요하지만, retryUntilSuccess에는 await이 없어도 되는지 설명해주실 수 있을까요?
```suggestion
    return retryUntilSuccess(callbackFn);
```

Reply:
1. callbackFn에는 await를 붙여야합니다.

왜냐하면 await를 try 내부의 callbackFn앞에 await를 붙이지 않으면 이 에러를 try가 캐치할 수 없기 때문입니다. 이는 JavaScript의 비동기 처리 방식과 관련이 있습니다. 비동기 처리는 실행 컨텍스트 스택(콜 스택)이 아니라, Web API같은 외부에서 처리가 됩니다. 또한 에러를 throw했을 때 이를 catch하기 위해서는 실행 컨텍스트 스택 기준 상위에 있어야합니다. 이 때 await를 비동기 함수 실행문 앞에 붙여줌으로써 해당 비동기 함수가 끝날 때까지 기다리도록 할 수 있습니다. 이러한 경우 비동기 함수는 동기적으로 처리가 되며 에러를 throw할 경우 try가 이 에러를 캐치할 수 있게 됩니다. 

정리하면, callbackFn 내의 비동기 함수에서 throw한 에러를 try가 catch하기 위해서는 await를 붙어주어야합니다.

2. 하지만 retryUntilSuccess 앞에는 await를 붙이지 않아도 됩니다.

이는 Promise객체의 특징과 관련이 있습니다. retryUntilSucess는 async를 앞에 붙였기 때문에 Promise객체를 반환합니다.(물론 InputView.getNames도 async함수이기에 Promise객체가 여기서도 반환합니다. 하지만 async로 Promise객체가 중첩되지는 않습니다.) 그리고 retryUntilSuccess를 호출하는 InputView.getNames에서는 retryUntilSuccess를 그대로 반환하고, InputView.getNames를 사용하는 CarRace.init()에서 await를 붙여주기 때문에 retryUntilSuccess는 await(기다리게)하게 됩니다.

Reply:
bf44a769dfdc4b851a958fc10ab0274498adf11c

Reply:
와우! 예시까지 정말 잘 정리해주셨네요! 👍👍👍